src/history/hs.c

```diff
@@ -894,9 +893,23 @@ __wt_find_hs_upd(
                  *
                  * We keep looking back into history store until we find a base update to apply the
                  * reverse deltas on top of.
+                 *
+                 * If our get key fails here we fall back to the datastore version. If its timestamp
+                 * doesn't match our timestamp then we've failed.
                  */
-                WT_ERR(hs_cursor->get_key(hs_cursor, &hs_btree_id, hs_key, &hs_start_tmp.timestamp,
-                  &hs_start_tmp.txnid, &hs_stop_tmp.timestamp, &hs_stop_tmp.txnid));
+                if ((ret =
+                        hs_cursor->get_key(hs_cursor, &hs_btree_id, hs_key, &hs_start_tmp.timestamp,
+                          &hs_start_tmp.txnid, &hs_stop_tmp.timestamp, &hs_stop_tmp.txnid)) != 0) {
+                    if (hs_stop_tmp.timestamp == on_disk_start->timestamp &&
+                      hs_stop_tmp.txnid == on_disk_start->txnid) {
```

Comment:
I think the hs_stop_tmp comes from the history store key. This is not valid after restart and there is an open ticket for that. Generally I think we should avoid using that.

Reply:
How do I avoid using it in this case?

Reply:
Removed txnid.